From f7c0f5fc1c619308139b2e4c099bc139ad45f7a2 Mon Sep 17 00:00:00 2001
From: Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>
Date: Mon, 7 May 2018 18:44:22 +0000
Subject: media router: Fix GCC build after 72d48c21c
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Commit 72d48c21c ("Add MRUIBase and MRViewsUI") added code that follows the
following pattern:

    base::Optional<T> Foo() {
      T t;
      return t;
    }

clang seems to automatically convert |t| to an rvalue reference, but GCC
fails with:

    ../../chrome/browser/ui/media_router/media_router_ui_base.cc: In member function ‘base::Optional<media_router::RouteParameters> media_router::MediaRouterUIBase::GetRouteParameters(const Id&, media_router::MediaCastMode)’:
    ../../chrome/browser/ui/media_router/media_router_ui_base.cc:325:10: error: could not convert ‘params’ from ‘media_router::RouteParameters’ to ‘base::Optional<media_router::RouteParameters>’
       return params;
              ^~~~~~

    ../../chrome/browser/ui/webui/media_router/media_router_ui.cc: In member function ‘base::Optional<media_router::RouteParameters> media_router::MediaRouterUI::GetLocalFileRouteParameters(const Id&, const GURL&, content::WebContents*)’:
    ../../chrome/browser/ui/webui/media_router/media_router_ui.cc:421:10: error: could not convert ‘params’ from ‘media_router::RouteParameters’ to ‘base::Optional<media_router::RouteParameters>’
       return params;
              ^~~~~~

Make the code build everywhere by explicitly using std::move() on the return
value.

Bug: 819294
Change-Id: I3c9d76d1ac595be90417603dcde9ff43c147fadd
Reviewed-on: https://chromium-review.googlesource.com/1043870
Reviewed-by: Derek Cheng <imcheng@chromium.org>
Commit-Queue: Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>
Cr-Commit-Position: refs/heads/master@{#556509}

diff --git a/chrome/browser/ui/media_router/media_router_ui_base.cc b/chrome/browser/ui/media_router/media_router_ui_base.cc
index 1412a60c9709..029ca9feaba0 100644
--- a/chrome/browser/ui/media_router/media_router_ui_base.cc
+++ b/chrome/browser/ui/media_router/media_router_ui_base.cc
@@ -322,7 +322,7 @@ base::Optional<RouteParameters> MediaRouterUIBase::GetRouteParameters(
   CHECK(initiator());
   params.incognito = initiator()->GetBrowserContext()->IsOffTheRecord();
 
-  return params;
+  return base::make_optional(std::move(params));
 }
 
 bool MediaRouterUIBase::CreateRoute(const MediaSink::Id& sink_id,
diff --git a/chrome/browser/ui/webui/media_router/media_router_ui.cc b/chrome/browser/ui/webui/media_router/media_router_ui.cc
index 140797e9742f..5791e3488f40 100644
--- a/chrome/browser/ui/webui/media_router/media_router_ui.cc
+++ b/chrome/browser/ui/webui/media_router/media_router_ui.cc
@@ -418,7 +418,7 @@ base::Optional<RouteParameters> MediaRouterUI::GetLocalFileRouteParameters(
   CHECK(initiator());
   params.incognito = initiator()->GetBrowserContext()->IsOffTheRecord();
 
-  return params;
+  return base::make_optional(std::move(params));
 }
 
 bool MediaRouterUI::ConnectRoute(const MediaSink::Id& sink_id,
-- 
2.14.1

